### PR TITLE
Ports Lusty Xenomorph Maid

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Port of: https://github.com/tgstation/-tg-station/pull/14947

Adds in the Lusty Xenomorph Maid

It's a Xeno....in a maid outfit....and it's freaking obsessed with cleaning things.

Can't do any damage or smash tables or anything---it just cleans blood and cleanable decals

Inaccessible and admin only, at the moment.

img
Thanks to @nienhaus for the sprites, of course!

:cl: Fox McCloud
add: Adds in the Lusty Xenomorph Maid Mob; currently admin only.
/:cl:

### CODE REVIEW

**Comment by KasparoVy** [comment]
Lmfao, that's wild. Lil' bit of fun

**Comment by tigercat2000** [comment]
As a player-accessible thing? Fuck no.
As an admin thing to spawn in, like ponies? Lusty maaaaaaaaaid.

**Comment by Necaladun** [comment]
As tiger said exactly.

We're really over saturated for silly memes in game.

**Comment by Fox-McCloud** [comment]
Ignore @tigercat2000 and @Necaladun

They're just part of the janitor's union and are shilling for their cause.

**Comment by Spacemanspark** [comment]
I dunno if we need something that originated from a story about alien porn.

**Comment by DaveTheHeadcrab** [comment]
@Spacemanspark 
You don't want to polish the Captain's laser gun?

**Comment by Spacemanspark** [comment]
I'll wreck you.

**Comment by DaveTheHeadcrab** [comment]
Dinner first, babe.

**Comment by Spacemanspark** [comment]
I'm broke.

**Comment by DaveTheHeadcrab** [comment]
Darn, I tried.

**Comment by Fox-McCloud** [comment]
if you ask nicely enough, @Spacemanspark, maybe they'll polish your monitor

**Comment by KasparoVy** [comment]
Oh God, hahaha: `visible_message("[src] polishes \the [target].")`

Excellent

File: code/modules/mob/living/simple_animal/hostile/alien.dm [hunk 1]
```diff
@@ -145,3 +145,25 @@
...
+
+/mob/living/simple_animal/hostile/alien/maid/AttackingTarget()
+	if(istype(target, /atom/movable))
+		if(istype(target, /obj/effect/decal/cleanable))
+			visible_message("[src] cleans up \the [target].")
+			qdel(target)
+			return
+		var/atom/movable/M = target
+		M.clean_blood()
+		visible_message("[src] polishes \the [target].")
```
Comment (tigercat2000) [inline code/modules/mob/living/simple_animal/hostile/alien.dm]: `"lusty xenomorph maid polishes Baldie McGriefer."`
improper capitalization, use either `\the [src]` or change the name to `Lusty Xenomorph maid`
also, lewd.

**Comment by NTSAM** [comment]
i love you

File: code/modules/mob/living/simple_animal/hostile/alien.dm [hunk 2]
```diff
@@ -145,3 +145,25 @@
...
+	friendly = "caresses"
+	environment_smash = 0
+	icon_state = "maid"
+	icon_living = "maid"
+	icon_dead = "maid_dead"
+
+/mob/living/simple_animal/hostile/alien/maid/AttackingTarget()
+	if(istype(target, /atom/movable))
+		if(istype(target, /obj/effect/decal/cleanable))
+			visible_message("[src] cleans up \the [target].")
```
Comment (Tastyfish) [inline code/modules/mob/living/simple_animal/hostile/alien.dm]: The soap item puts cleaning messages in `<span class='notice'>`'s.

File: code/modules/mob/living/simple_animal/hostile/alien.dm (hunk 1 above)
Comment (Tastyfish) [inline code/modules/mob/living/simple_animal/hostile/alien.dm]: `\The`* if you're trying to get capitalization right

**Comment by Necaladun** [comment]
I'll polish your [target] ;)

More seriously though, this is yet another meme-based thing I'd prefer not to see player accessible. 

We're near saturation point for them, and I'd prefer the room for sillymemeness be left for well...better memes.

**Comment by tigercat2000** [comment]
I don't like it entirely on the basis of it's origin point, a joke lewd book.

**Comment by Necaladun** [comment]
Thats kinda what I meant by leave room for better memes.

At least it's not a WGW reference, but still...

**Comment by Spacemanspark** [comment]
Oh don't worry @Necaladun I'm sure /tg/ has one of those too.

**Comment by Fox-McCloud** [comment]
@Spacemanspark 

We already have those references, sorry, too late for you ;)

**Comment by Mijhead** [comment]
I don't think this is really necessary, for the reasons that @Necaladun gave. Besides that, just because there's a preceding thing, such as the librarian's PDA model number, doesn't mean that it's okay to add similar things.

**Comment by Aurorablade** [comment]
i actually support it as an admin spawn.

**Comment by fludd12** [comment]
I support this as-is.

Besides, xenobio gets filthy in my presence, and I need someone to clean it up with me

;)

**Comment by Robertha89** [comment]
Yes, this is great only grouchy people don't want this, people who are against fun!

**Comment by Magikarpet** [comment]
Why not give xenobiologists something comedic to play with? Not eveything that's hilarious/a reference should be locked behind admin-spawns. :+1:

**Comment by tigercat2000** [comment]
@Magikarpet References to lewd stories, probably should.

**Comment by Earthdivine** [comment]
I definatly would rather not have player spawnable. We have had people get in trouble for having the dang book in the first place. This will only make things worse.

**Comment by TheDZD** [comment]
No longer player accessible.

**Pull request merged**